praw/util/listindex.py

```diff
@@ -0,0 +1,20 @@
+"""Implement a function similar to dict.get(item, default)."""
+
+from typing import Any, List, Optional
+
+
+def index(
+    retrieval_list: List[Any], index_number: int, default: Optional[Any] = None
+) -> Any:
```

Comment:
Two comments on this function:

1. The naming conflicts with the existing [`.index()` function on sequences](https://docs.python.org/3/library/stdtypes.html#typesseq-common), which gives the index of its argument. A better name would be something along the lines of "get or default."
2. I think this function makes code harder to understand, because it requires the reader to go look up a separate function to understand the code at hand. In addition, you use it in one place right now. I believe the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) applies — once this pattern has been duplicated in three places, then it makes sense to refactor the common functionality into a helper function.